tests/testenv/processes/postgresql.py

```diff
@@ -0,0 +1,54 @@
+########
+# Copyright (c) 2013 GigaSpaces Technologies Ltd. All rights reserved
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+#    * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    * See the License for the specific language governing permissions and
+#    * limitations under the License.
+
+import os
+import logging
+
+from cloudify.utils import setup_logger
+
+from .. import utils
+
+
+logger = setup_logger('postgresql')
+PS_SERVICE_NAME = 'postgresql'
+
+
+class Postgresql(object):
+    """
+    Manages an PostgreSQL lifecycle.
+    """
+
+    def __init__(self):
+        self._on_ci = os.environ.get('CI') == 'true'
+        self._ps_port = 5432
+        setup_logger('postgresql', logging.INFO)
+        setup_logger('postgresql.trace', logging.INFO)
+
+    def is_running(self):
+        return utils.is_port_open(self._ps_port)
+
+    def configure(self, db_name, username, password):
+        if not self.is_running():
+            logger.warning('postgresql service is not running')
+            return
+        logger.info('Configuring postgresql db..')
+        ps_config_source = 'tests/postgresql_configuration.sh'
+        ps_config_destination = '/tmp/postgresql_configuration.sh'
+        utils.copy(source=ps_config_source,
+                   destination=ps_config_destination)
+        utils.sudo('chmod +x {0}'.format(ps_config_destination))
```

Comment:
Do you really have to use sudo?
It means that test will have to run with sudo permissions, which is not good...